app/views/videos/show.html.haml

```diff
@@ -2,14 +2,14 @@
   .container
     .row
       .video_large_cover.col-sm-7.col-sm-offset-1
-        %img(src='/tmp/#{@video.large_cover_url}')
+        %img(src='#{@video.large_cover}')
       .video_info.col-sm-3
         %header
           %h3= @video.title
           %span Rating: 4.5/5.0
         %p= @video.description
         .actions
-          %a.btn.btn-primary(href="") Watch Now
+          %a.btn.btn-primary(href='#{@video.video_url}') Watch Now
```

Comment:
You should use `link_to` Rails helper for better Readability. At this stage you should know how it works. For instance on line 13 - you are using it..